From c057671a16cc685e59b7ae8ab14a2c37b9c56ff8 Mon Sep 17 00:00:00 2001
From: Alessio Igor Bogani <alessio.bogani@elettra.eu>
Date: Sun, 3 May 2026 15:05:43 +0200
Subject: [PATCH 2/2] powerpc: Add __feclearexcept and __fetestexcept internal
 aliases for e500

Apply the equivalent of upstream commits 2eb8836de7 ("math: Add
feclearexcept internal alias") and 63ba1a1509 ("math: Add fetestexcept
internal alias") to the e500 sysdeps, which were missed because the SPE
support had been removed when those changes landed.  Without these
aliases the build fails with "aliased to undefined symbol
'__feclearexcept'" once the SPE port is restored.
---

Upstream-Status: Inappropriate

 sysdeps/powerpc/powerpc32/e500/nofpu/fclrexcpt.c   | 1 +
 sysdeps/powerpc/powerpc32/e500/nofpu/ftestexcept.c | 4 +++-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/sysdeps/powerpc/powerpc32/e500/nofpu/fclrexcpt.c b/sysdeps/powerpc/powerpc32/e500/nofpu/fclrexcpt.c
index 3e783e2c2f..db3af372ab 100644
--- a/sysdeps/powerpc/powerpc32/e500/nofpu/fclrexcpt.c
+++ b/sysdeps/powerpc/powerpc32/e500/nofpu/fclrexcpt.c
@@ -35,19 +35,20 @@ __feclearexcept (int excepts)
   fesetenv_register (fpescr);
 
   /* Let the kernel know if the "invalid" or "underflow" bit was
      cleared.  */
   if (excepts & (FE_INVALID | FE_UNDERFLOW))
     __fe_note_change ();
 
   /* Success.  */
   return 0;
 }
+libm_hidden_def (__feclearexcept)
 
 #include <shlib-compat.h>
 #if SHLIB_COMPAT (libm, GLIBC_2_1, GLIBC_2_2)
 strong_alias (__feclearexcept, __old_feclearexcept)
 compat_symbol (libm, __old_feclearexcept, feclearexcept, GLIBC_2_1);
 #endif
 
 libm_hidden_ver (__feclearexcept, feclearexcept)
 versioned_symbol (libm, __feclearexcept, feclearexcept, GLIBC_2_2);
diff --git a/sysdeps/powerpc/powerpc32/e500/nofpu/ftestexcept.c b/sysdeps/powerpc/powerpc32/e500/nofpu/ftestexcept.c
index 57d763c41d..aaeae0581b 100644
--- a/sysdeps/powerpc/powerpc32/e500/nofpu/ftestexcept.c
+++ b/sysdeps/powerpc/powerpc32/e500/nofpu/ftestexcept.c
@@ -12,20 +12,22 @@
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
    Lesser General Public License for more details.
 
    You should have received a copy of the GNU Lesser General Public
    License along with the GNU C Library; if not, see
    <https://www.gnu.org/licenses/>.  */
 
 #include <fenv_libc.h>
 
 int
-fetestexcept (int excepts)
+__fetestexcept (int excepts)
 {
   unsigned long f;
 
   /* Get the current state.  */
   f = fegetenv_register ();
 
   return __fexcepts_from_spe (f) & excepts;
 }
+libm_hidden_def (__fetestexcept)
+weak_alias (__fetestexcept, fetestexcept)
 libm_hidden_def (fetestexcept)
-- 
2.43.0

